app/workers/subject_metadata_worker.rb

```diff
@@ -0,0 +1,24 @@
+class SubjectMetadataWorker
+  include Sidekiq::Worker
+
+  def perform(set_id)
+
+    smses = SetMemberSubject.where(subject_set_id: set_id)
```

Comment:
you can remove this entirely, the `subject_set_id` is a FK relation on the `set_member_subjects` resource